From stable-bounces@linux.kernel.org Wed Oct 11 23:00:27 2006
Date: Thu, 12 Oct 2006 07:59:47 +0200
From: Jens Axboe <jens.axboe@oracle.com>
To: stable@kernel.org
Message-ID: <20061012055946.GU6515@kernel.dk>
Mime-Version: 1.0
Content-Disposition: inline
Subject: splice: fix pipe_to_file() ->prepare_write() error path
Content-Type: text/plain; charset="us-ascii"

From: Jens Axboe <jens.axboe@oracle.com>

Don't jump to the unlock+release path, we already did that.

Signed-off-by: Jens Axboe <jens.axboe@oracle.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 fs/splice.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- linux-2.6.18.1.orig/fs/splice.c
+++ linux-2.6.18.1/fs/splice.c
@@ -607,7 +607,7 @@ find_page:
 			ret = -ENOMEM;
 			page = page_cache_alloc_cold(mapping);
 			if (unlikely(!page))
-				goto out_nomem;
+				goto out_ret;
 
 			/*
 			 * This will also lock the page
@@ -666,7 +666,7 @@ find_page:
 		if (sd->pos + this_len > isize)
 			vmtruncate(mapping->host, isize);
 
-		goto out;
+		goto out_ret;
 	}
 
 	if (buf->page != page) {
@@ -698,7 +698,7 @@ find_page:
 out:
 	page_cache_release(page);
 	unlock_page(page);
-out_nomem:
+out_ret:
 	return ret;
 }
 
